hnn_core/viz.py

```diff
@@ -577,12 +577,15 @@ def plot_cells(net, ax=None, show=True):
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from mpl_toolkits.mplot3d import Axes3D  # noqa: F401 unused import
+    from mpl_toolkits.mplot3d import Axes3D
 
     if ax is None:
         fig = plt.figure()
         ax = fig.add_subplot(111, projection='3d')
 
+    elif not isinstance(ax, Axes3D):
+        raise TypeError("The 'ax' argument must be an instance of Axes3D.")
```

Comment:
when raising error messages, it is good practice to say what you got:

"Expected blah, got blahblah"

You save the user some guesswork by doing this ... also helps users without advanced debugging capabilities